pkg/server/event_handler.go

```diff
@@ -293,6 +305,46 @@ func (h *EventHandler[T, P]) getGCSObjectProto(ctx context.Context, objAttrs map
 	return p, nil
 }
 
+type payloadMetadata struct {
+	Metadata map[string]string
+}
+
+func parseGitHubSource(ctx context.Context, data []byte) (*v1alpha1.GitHubSource, error) {
+	pm := &payloadMetadata{}
+	if err := json.Unmarshal(data, pm); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal payloadMetadata %w", err)
+	}
+
+	r := &v1alpha1.GitHubSource{}
+
+	c, found := pm.Metadata["git-commit"]
+	if found {
+		r.Commit = c
+	}
+	r.Commit = c
+	rn, found := pm.Metadata["git-repo"]
+	if found {
+		r.RepoName = rn
+	}
+	w, found := pm.Metadata["git-workflow"]
+	if found {
+		r.Workflow = w
+	}
+	ws, found := pm.Metadata["git-workflow-sha"]
+	if found {
+		r.WorkflowSha = ws
+	}
+	t, found := pm.Metadata["triggered-timestamp"]
+	if found {
+		date, err := time.Parse("2006-01-02T15:04:05", t)
```

Comment:
Should we use time.RFC3339 for the format instead?

Reply:
Currently the timestamp format is defined here: https://github.com/abcxyz/pmap/blob/main/.github/workflows/snapshot-file-change.yml#L35.

It has format of `2006-01-02T15:04:05`, but time.RFC3339 requires format of `2006-01-02T15:04:05Z07:00`

This requires modification to workflow files, I think we can fix this with #94

Reply:
Added to use time.RFC3339